bitbots_head_behavior/src/bitbots_head_behavior/actions/visual_kompass_record.py

```diff
@@ -0,0 +1,73 @@
+import math
+
+import rospy
+
+from std_msgs.msg import Header
+
+from dynamic_stack_decider.abstract_action_element import AbstractActionElement
+
+
+class VisualKompassRecord(AbstractActionElement):
+    """
+    Executes the configured pattern the scan the ground truth for the Visual Kompass.
+    """
+
+    def __init__(self, blackboard, dsd, parameters=None):
+        super(VisualKompassRecord, self).__init__(blackboard, dsd, parameters)
+        
+        self.index = 0
+        self.pan_speed = 1.0
+        self.tilt_speed = 1.0
+        self.interpolation_steps = 4
+        self.pattern_left = self.blackboard.head_capsule.generate_pattern(  self.blackboard.config['record_pattern_scan_lines'],
+                                                                            max(self.blackboard.config['record_pattern_pan_max_left']),
+                                                                            min(self.blackboard.config['record_pattern_pan_max_left']),
+                                                                            max(self.blackboard.config['record_pattern_tilt_max']),
+                                                                            min(self.blackboard.config['record_pattern_tilt_max']),
+                                                                            interpolation_steps=self.interpolation_steps)
+
+        self.pattern_right = self.blackboard.head_capsule.generate_pattern(  self.blackboard.config['record_pattern_scan_lines'],
+                                                                            max(self.blackboard.config['record_pattern_pan_max_right']),
+                                                                            min(self.blackboard.config['record_pattern_pan_max_right']),
+                                                                            max(self.blackboard.config['record_pattern_tilt_max']),
+                                                                            min(self.blackboard.config['record_pattern_tilt_max']),
+                                                                            interpolation_steps=self.interpolation_steps)
+
+        self.visual_compass_trigger = rospy.Publisher( self.blackboard.config['vc_trigger_topic'], Header, queue_size=5)
```

Comment:
This should be moved to `head_node.py` where the other publishers are registered, too.